float64 a
float64 b
float64 c
float64 vx 
float64 vy 
float64 vz
float64 wx
float64 wy
float64 wz

---

float64[] q_dot
